src/components/actions/evaluatorFunctions.ts

```diff
@@ -8,6 +8,42 @@ Any changes done here should also be replicated in front-end
 
 import { DateTime, Duration } from 'luxon'
 
+type FilterRule = 'exclude' | 'include'
+interface FilterOptions {
+  key?: string
+  rule?: FilterRule
+  values: string | string[]
+}
+
+const filterArray = (valuesArray: Object[] | string[], options: FilterOptions) => {
```

Comment:
I don't think it should be restricted to just objects and strings -- any primitive type should work fine.

Reply:
This is a good case for using `valuesArray: unknown[]` and then having proper type guards within (as suggested above).